src/components/Tabs/index.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { Box, Tab, type BoxProps, type TabProps } from '@mui/material';
+import { TabContext, TabList, TabPanel, type TabPanelProps } from '@mui/lab';
+
+export interface TabsProps {
+	tabs: Array<{
+		label: string;
+		render: JSX.Element;
+		tabProps?: TabProps;
+		tabPanelProps?: TabPanelProps;
+	}>;
+	tabsListBoxProps?: BoxProps;
+	defaultTab?: string;
+	onTabChange?: (newTab: string) => void;
+}
+
+export const Tabs = ({
+	tabs,
+	tabsListBoxProps,
+	defaultTab,
+	onTabChange,
+}: TabsProps) => {
+	const [tab, setTab] = React.useState(
+		defaultTab != null
+			? tabs.find(({ label }) => label === defaultTab)?.label ?? tabs[0].label
+			: tabs[0].label,
+	);
+
+	if (!tabs.length) {
+		return null;
+	}
+
+	const handleChange = (_event: React.SyntheticEvent, newValue: string) => {
+		setTab(newValue);
+		onTabChange?.(newValue);
+	};
+
+	const { sx, ...boxProps } = tabsListBoxProps ?? {};
+
+	return (
+		<TabContext value={tab}>
+			<Box
+				sx={[
+					{ borderBottom: 1, borderColor: 'divider' },
+					// See: https://mui.com/system/getting-started/the-sx-prop/#passing-the-sx-prop
+					...(Array.isArray(sx) ? sx : [sx]),
```

Comment:
Not for this PR but it might be useful at some point to move this to a util function (I see already 2 references in balena-ui)

Reply:
+1